### PR TITLE
Refactor the operator code to ease control-plane testing

### PR DESCRIPTION
**Pull request by pippolo84**
The current structure of the operator makes impossible to use it in control-plane unit test framework. Specifically, the 3 main blockers are:

1. all the code in the `main` package under the `operator` folder cannot be imported in unit tests
2. the use of the RESTClient does not enable events injection through "fake" generated k8s clients
3. the use of a Viper singleton and a registered options map singleton make impossible to have both an agent and an operator running in memory in the same control-plane test binary

The PR addresses those issues with the following approach:

1. move the operator code in a separate `cmd` package, replicating the same structure of the daemon
2. introduce the use of ListerWatcher from typed k8s clients
3. avoid the use of the Viper singleton in favor of a per-component Viper (`vp`) and registered options map (`regOpts`) instances

Though the number of changes introduced by the PR is high, they are mostly mechanical substitutions and/or code moving, thus the review process should be doable.

### CODE REVIEW

**Review by christarazi — APPROVED** [review]

**Review by joamaki — APPROVED** [review]
Looks reasonable to me and it'll make my modular config PR (https://github.com/cilium/cilium/pull/20829) a bit cleaner as well!

File: cilium/cmd/cleanup.go [hunk 1]
```diff
@@ -40,6 +40,9 @@ working condition after uninstalling the Cilium agent.`,
 }
 
 var (
+	vp      *viper.Viper              = viper.New()
+	regOpts *option.RegisteredOptions = option.NewRegisteredOptions(vp)
+
```
Comment (tklauser) [inline cilium/cmd/cleanup.go]: Nit: these don't seem to be specific for the `cleanup` command (i.e. the `cleanup.go`) but used across several commands in the package. Maybe move them to `root.go`?

Comment (pippolo84) [reply cilium/cmd/cleanup.go]: That´s right, I moved it.

**Review by ldelossa — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Pull request merged**